rest/server_context.go

```diff
@@ -206,10 +206,17 @@ func (sc *ServerContext) GetDatabase(ctx context.Context, name string) (*db.Data
 		return nil, base.HTTPErrorf(http.StatusBadRequest, "invalid database name %q", name)
 	}
 
+	dbc, err := sc.unsuspendDatabase(ctx, name)
+	if err != nil && err != base.ErrNotFound && err != ErrSuspendingDisallowed {
+		return nil, err
+	} else if err == nil {
+		return dbc, nil
+	}
+
+	// database not loaded, fallback to fetching it from cluster
+
 	if sc.BootstrapContext.Connection != nil {
-		sc.lock.Lock()
-		defer sc.lock.Unlock()
-		// database not loaded, go look for it in the cluster
+		var found bool
 		found, err := sc._fetchAndLoadDatabase(ctx, name)
```

Comment:
There is still code missing from this function. 

Have a look at `GetDatabase` in this commit to see the missing code: https://github.com/couchbase/sync_gateway/commit/82bbfee5cb26d44585dc1c38c31d72b643f396c6

Reply:
I think I got this now.